platform/jobs/RetirementJobs.groovy

```diff
@@ -152,10 +152,20 @@ job('user-retirement-collector') {
     // generally there are always available jenkins-worker instances idling.
     label('jenkins-worker')
 
-    // Disallow this job to have simultaneous instances building at the same
-    // time.  This would prevent race conditions related to triggering multiple
-    // retirement driver jobs against the same user.
-    concurrentBuild(false)
+    // Allow this job to have simultaneous instances building at the same time
+    // in general, but use the throttle-concurrents plugin to limit only one
+    // instance of this job to run concurrently per environment.  This would
+    // prevent race conditions related to triggering multiple retirement driver
+    // jobs against the same user in the same environment.
+    concurrentBuild(true)
+    throttleConcurrentBuilds {
+        maxTotal(1)
```

Comment:
I think specifying a maxTotal here will negate the paremeter throttling below. You should be able to just use parameter throttling.